### PR TITLE
Blocks/HTTP: ServeStaticFilesFrom: Fixed slash in subdirectory, added tests.

### PR DESCRIPTION
**Pull request by sompylasar**
Unfortunately, a bug leaked through, wasn't covered by the tests.

CC @dkorolev

### CODE REVIEW

File: Blocks/HTTP/impl/posix_server.h [hunk 1]
```diff
@@ -262,7 +262,8 @@ class HTTPServerPOSIX final {
           if (!content_type.empty()) {
             // `url_dirname` has no trailing slash.
             const std::string url_dirname =
-                options.url_base + current::strings::Join(item_info.path_components_cref, '/');
+                options.url_base + (options.url_base == "/" || item_info.path_components_cref.empty() ? "" : "/") +
```
Comment (dkorolev) [inline Blocks/HTTP/impl/posix_server.h]: Extra parentheses to stress the `||` is to the left of `?` would help here. LGTM! Feel free to merge yourself, making you an admin now.

Comment (sompylasar) [inline Blocks/HTTP/impl/posix_server.h]: Thanks! Yes, will add ()

Comment (dkorolev) [inline Blocks/HTTP/impl/posix_server.h]: :+1:

**Comment by dkorolev** [comment]
Last three commits LGTM. Thanks @sompylasar!

**Pull request merged**